### ISSUE TITLE
Indentation rules not working correctly?

### ISSUE THREAD

**Issue by jefflevinson** [microsoft/monaco-editor#86]
Hopefully this is just me, but using the playground I added this:

```
monaco.languages.setLanguageConfiguration('mySpecialLanguage', {
    indentationRules: {
        decreaseIndentPattern: /(| project)/,
        increaseIndentPattern: /(| LinkedWorkItems)/,
        indentNextLinePattern: null,
        unIndentedLinePattern: null
    }
});
```

When I enter a "| LinkedWorkItems" it correctly indents. However, at no time afterwards does entering "| project" outdent the indentation again. Did I miss something here? I can't figure out how to even verify what text pattern it is trying to match against. I tried using tabs, spaces, heck I even went to .*| project to no avail.

**Comment by jefflevinson** [comment]
One other note - I tried this rule:
monaco.languages.setLanguageConfiguration('mySpecialLanguage', {
    indentationRules: {
        decreaseIndentPattern: /(\| project .*)/,
        increaseIndentPattern: /(\| LinkedWorkItems)/,
        indentNextLinePattern: /(\| project .*)/,
        unIndentedLinePattern: null
    }
});

And it did indeed indent a further level after the | project abcdef line and then outdented. So I know it's recognizing the | project .* pattern but it just seems like the decreaseIndentPattern rule is not working. If I set it to \w* it does decrease the indent but obviously that pattern is too broad for use. I just can't figure out where/how it is interpreting this rule to see if it's an actual bug.
